registry/handlers/hooks.go

```diff
@@ -0,0 +1,103 @@
+package handlers
+
+import (
+	"fmt"
+	"net/smtp"
+	"strings"
+
+	"github.com/Sirupsen/logrus"
+	"github.com/docker/distribution/configuration"
+)
+
+// ReportHook is for hooking Panic in web application
+type ReportHook struct {
+}
+
+// Fire forwards an error to ReportHook
+func (hook *ReportHook) Fire(entry *logrus.Entry) error {
+	logrus.Info("Message: ", entry.Message)
+	if errConfig.Log.Handlers.Errors.Type == "email" {
```

Comment:
No need to check this: make a hook that is configured when the "email" type is seen.